sstables/sstable_set.cc

```diff
@@ -356,13 +356,25 @@ bool partitioned_sstable_set::insert(shared_sstable sst) {
         sub_bytes_on_disk(n);
     });
 
+    auto maybe_insert_run_fragment = [this] (const shared_sstable& sst) mutable {
+        auto it = _all_runs.find(sst->run_identifier());
+        if (it == _all_runs.end()) {
+            auto new_run = make_lw_shared<sstable_run>();
+            if (!new_run->insert(sst)) {
+                on_internal_error(sstlog, fmt::format("Empty SSTable run should not have rejected {}", sst->get_filename()));
+            }
+            return _all_runs.emplace(sst->run_identifier(), std::move(new_run)).second;
+        }
+        return it->second->insert(sst);
+    };
```

Comment:
I don't understand this change in the context of this patch.

Reply:
Maybe it should be in the next patch?

Reply:
The reason I introduced it here is because _all_runs's value is now a lw_shared_ptr of sstable_run. So when inserting a new value into map, I have to emplace a new ptr with a run containing the single element. And it preserves the strong exception safety guarantee in the function.